attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -37,20 +68,20 @@ impl Attester for TdxAttester {
 
         report_data.resize(64, 0);
 
-        let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
-            d: report_data.as_slice().try_into()?,
-        };
+        let quote_bytes = match TsmReportPath::open() {
+            Ok(tsm) => {
+                let report = tsm.attestation_report(TsmReportProvider::Tdx(report_data));
+                tsm.close();
 
-        let engine = base64::engine::general_purpose::STANDARD;
-        let quote = match tdx_attest_rs::tdx_att_get_quote(Some(&tdx_report_data), None, None, 0) {
-            (tdx_attest_rs::tdx_attest_error_t::TDX_ATTEST_SUCCESS, Some(q)) => engine.encode(q),
-            (error_code, _) => {
-                return Err(anyhow!(
-                    "TDX Attester: Failed to get TD quote. Error code: {:?}",
-                    error_code
-                ));
+                match report {
+                    Ok(bytes) => bytes,
+                    Err(e) => bail!("TDX Attester: {}", e),
+                }
             }
+            Err(e) => try_getquote_ioctl_or_fail(e, report_data)?,
```

Comment:
I don't like passing `e` into this function. Instead you could probably do something like `Err(e1) => try_getquote().map_err(|e2| anyhow!("{} {}",e1,e2)?`